lib/new_relic/agent/instrumentation/custom_events_subscriber.rb

```diff
@@ -0,0 +1,37 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'new_relic/agent/instrumentation/notifications_subscriber'
+
+# Listen for ActiveSupport::Notifications events for custom events
+module NewRelic::Agent::Instrumentation
+  class CustomEventsSubscriber < NotificationsSubscriber
+    def start(name, id, _payload) # THREAD_LOCAL_ACCESS
+      return unless state.is_execution_traced?
+
+      finishable = NewRelic::Agent::Tracer.start_transaction_or_segment(name: transaction_name(name),
+        category: :custom_events)
+      push_segment(id, finishable)
+    rescue => e
+      log_notification_error(e, name, 'start')
+    end
+
+    def finish(name, id, payload) # THREAD_LOCAL_ACCESS
+      return unless state.is_execution_traced?
+
+      NewRelic::Agent.notice_error(payload[:exception_object]) if payload.key?(:exception_object)
+
+      finishable = pop_segment(id)
+      finishable.finish if finishable
+    rescue => e
+      log_notification_error(e, name, 'finish')
+    end
+
+    private
+
+    def transaction_name(name)
+      "Controller/ActiveSupport/CustomEvents/#{name}"
```

Comment:
It's such a bummer we have to start these with `Controller/`. Does fixing that seem like a 9.0-worthy change?

Reply:
We should definitely gather some feedback on naming conventions on the whole to better serve the (current) Ruby community, and then gate anything significant with a major version bump.

For this feature here, it's following the `Controller/` pattern used by all other notification instrumentation. I suppose we could use `Custom/` instead. Do you want to have us think more on naming for this feature specifically or hold off for the bigger version-bump-requiring conversation?

Reply:
I thought `Controller/` was a magic prefix that we needed to use in order for the action to be permitted as the root node of a transaction? I'll poke around the codebase to check that assertion.

If `Controller/*` isn't a magic required prefix, I think it'd be worth discussing what we want to name the transactions since we're creating a new type of transaction. I'd prefer to get it right before release rather than change it on customers in a major version.

Reply:
Hmmm. I'm not 100% sure. The magic prefix thing DOES seem familiar to me. The agent might need it, or maybe the UI needs it? Or both or neither, I'm not sure. Maybe we could test out changing it to something else and seeing how it ends up looking in the UI?

Reply:
Assuming we aren't tied to `Controller/`, I do like the `Custom/`prefix better. I think it gives a nice indicator of what a user is looking at/ that someone on their team found a particular metric important to capture, and leads down a clearer path that those metrics are customizable.

Reply:
I like the idea of testing it in the UI. I'm hesitant about using the `Custom/` prefix because that is used by customers creating custom metrics: 
https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent.rb#L183 
https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent.rb#L209

I think my preference would be to start it with `ActiveSupport/`.

Reply:
Re: `Controller/*`, I think this came up while we were adding gRPC instrumentation: https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent/instrumentation/grpc/server/instrumentation.rb#L115 
I can't tell from the commit why it's named this way, and the agent spec doesn't have this requirement.

Reply:
It could be due to this method: https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent/rules_engine.rb#L26-L41

Reply:
It looks like the existing Rails notifications transactions don't all begin with `Controller/`  (especially given that we have a view one that correctly starts with `View/`) so there's no need to hold on to `Controller/` for grouping all of a customer's notifications transactions together as I had originally thought.

I like Kayla's take on disambiguating a Rails custom event from our own agent's custom events and not using `Custom/` which is reserved for the latter.

I'll put together Kayla's recommended change:

```diff
- "Controller/ActiveSupport/CustomEvents/#{name}"
+ "ActiveSupport/CustomEvents/#{name}"
```

and get a UI test going to see what we think.

Reply:
Thank you, James!

Reply:
The recommended change looks great to me in the UI, with the transactions appearing as non-web when recorded on their own and not segments belonging to a web transaction.

<img width="753" alt="Screen Shot 2022-11-30 at 9 41 44 PM" src="https://user-images.githubusercontent.com/121392/204975579-212b14a8-48f2-4708-85c9-efdee70b630f.png">